src/app/page.tsx

```diff
@@ -1,36 +1,33 @@
 import Link from "next/link";
 
-import { LatestPost } from "~/app/_components/post";
-import { getServerAuthSession } from "~/server/auth";
 import { api, HydrateClient } from "~/trpc/server";
+import { getServerAuthSession } from "~/server/auth";
 import { UserProfile } from "./_components/userProfile/UserProfile";
 
 export default async function Home() {
-  const hello = await api.post.hello({ text: "from tRPC" });
   const session = await getServerAuthSession();
 
-  void api.post.getLatest.prefetch();
-
   if (!session) {
     return (
       <div>
         <p>
-          Welcome to <code>trpc</code> with <code>next-auth</code>!
-        </p>
-        <p>
-          <Link href="/api/auth/signin">Sign in</Link> to see the latest post.
+          <Link href="/api/auth/signin">Sign in</Link>
         </p>
       </div>
     );
   }
 
+  const team = await api.team.getByUserId({userId: session.user.id})
+
+  if (!team) return <h1>Error: No team found for user</h1>
```

Comment:
NIT: This can be made more concise and React-y using the logical `&&` or  ternary `?`. If more conditions pop up we can use switch or if statements.